search.py

```diff
@@ -513,25 +514,20 @@ def __init__(self, initial= None, goal= None, size = 3, heuristic = 'manhattan_d
         assert self.check_solvability(initial), "The puzzle is not solvable from the given initial state!"
         Problem.__init__(self, initial, goal)
     def shuffle_state(self, shuffle, state):
-        """ perfrom random action from the initial state"""
-
+        """ perform random action from the initial state"""
```

Comment:
Comments should start with a capital letter. Also, comments should be all formatted the same way. Either leave a space around the comment (inside the string) from both sides, or keep it without spaces on the sides. Here you can either remove the starting space or add a space at the end of the sentence.